metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/ExecutionAggregator.java

```diff
@@ -1,56 +1,115 @@
  * @param <E> The execution metric to aggregate from the Execution
  */
-public interface ExecutionAggregator<T extends Execution, M, E> {
+public abstract class ExecutionAggregator<T extends Execution, M extends Metric, E> {
 
     /**
      * Get the metric to aggregate from a single workflow execution.
      * @param execution
      * @return
      */
-    E getMetricFromExecution(T execution);
+    public abstract E getMetricFromExecution(T execution);
 
     /**
-     * Aggregates TaskExecutions that belong to a single workflow run into a workflow-level RunExecution
-     * @param taskExecutionsForOneWorkflowRun
+     * Get the aggregated metric associated with the metric type from the aggregated Metrics class, which contains multiple types of aggregated metrics.
+     * @param metrics
      * @return
      */
-    Optional<T> getWorkflowExecutionFromTaskExecutions(TaskExecutions taskExecutionsForOneWorkflowRun);
+    public abstract M getMetricFromMetrics(Metrics metrics);
+
+    /**
+     * Get the executions containing the metric to aggregate from ExecutionsRequestBody.
+     * @param executionsRequestBody
+     * @return
+     */
+    public abstract List<T> getExecutionsFromExecutionRequestBody(ExecutionsRequestBody executionsRequestBody);
+
+    /**
+     * Returns a boolean indicating if the execution metric is valid.
+     * @param executionMetric
+     * @return
+     */
+    public abstract boolean validateExecutionMetric(E executionMetric);
 
     /**
      * Aggregates workflow executions into an aggregated metric.
-     * @param executions
+     * @param executionMetrics
      * @return
      */
-    Optional<M> getAggregatedMetricFromExecutions(List<T> executions);
+    protected abstract Optional<M> calculateAggregatedMetricFromExecutionMetrics(List<E> executionMetrics);
 
     /**
      * Aggregates a list of aggregated metrics into one aggregated metric.
      * @param aggregatedMetrics
      * @return
      */
-    Optional<M> getAggregatedMetricsFromAggregatedMetrics(List<M> aggregatedMetrics);
+    protected abstract Optional<M> calculateAggregatedMetricFromAggregatedMetrics(List<M> aggregatedMetrics);
+
+    /**
+     * Aggregate metrics from all submissions in the ExecutionsRequestBody.
+     * @param allSubmissions
+     * @return
+     */
+    public abstract Optional<M> getAggregatedMetricFromAllSubmissions(ExecutionsRequestBody allSubmissions);
 
     /**
-     * Returns a list of executions of type T where the execution metric is not null.
+     * Aggregates workflow executions into an aggregated metric and calculates the number of skipped executions.
      * @param executions
      * @return
      */
-    default List<E> getNonNullMetricsFromExecutions(List<T> executions) {
-        return executions.stream()
-                .map(this::getMetricFromExecution)
+
+    public final Optional<M> getAggregatedMetricFromExecutions(List<T> executions) {
```

Comment:
Method that calculates the aggregated metric plus sets the number of executions skipped.